crates/rpc/rpc/src/eth/cache/mod.rs

```diff
@@ -431,7 +480,10 @@ where
                         }
                         CacheAction::CacheNewCanonicalChain { blocks, receipts } => {
                             for block in blocks {
-                                this.on_new_block(block.hash, Ok(Some(block.unseal())));
+                                this.on_new_block(
+                                    block.hash,
+                                    Ok(Some(BlockData::Block(block.unseal()))),
```

Comment:
this needs to insert the BlockWithsenders,

we need to update:

https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc/src/eth/cache/mod.rs#L534-L534